Automattic/wp-calypso
# Update docs link in the domain transfer status icon

Closes: https://github.com/Automattic/wp-calypso/issues/86314

## Proposed Changes

This PR updates the invalid link `https://wordpress.com/support/domains/incoming-domain-transfer/#pending` that does not do the page jump to the right section on the page to `https://wordpress.com/support/domains/incoming-domain-transfer/#step-4-check-the-transfer-status` when checking the domain transfer status.

## Testing Instructions



* Pull the changes from this branch
* Apply the diff below:

```diff --git a/packages/domains-table/src/utils/resolve-domain-status.tsx b/packages/domains-table/src/utils/resolve-domain-status.tsx
index 6b2d3868389..e46149dc906 100644
--- a/packages/domains-table/src/utils/resolve-domain-status.tsx
+++ b/packages/domains-table/src/utils/resolve-domain-status.tsx
@@ -88,9 +88,9 @@ export function resolveDomainStatus(
                                />
                        ),
                },
-               args: {
-                       transferFinishDate: moment.utc( domain.transferEndDate ).format( 'LL' ),
-               },
+               // args: {
+               //      transferFinishDate: moment.utc( domain.transferEndDate ).format( 'LL' ),
+               // },
        };
 
        const mappingSetupCallToAction = {
@@ -172,6 +172,10 @@ export function resolveDomainStatus(
                                statusClass: 'status-success',
                                status: translate( 'Active' ),
                                icon: 'check_circle',
+                               noticeText: translate(
+                                       'Transfer failed. Learn the possible {{a}}reasons why{{/a}}.',
+                                       transferOptions
+                               ),
                        };
```
* Navigate to `http://calypso.localhost:3000/domains/manage`
* Please note that we temporarily added the link to the `Active` status of the domain in the diff above to be able to test the link easier (otherwise we would need to test with domain transfer which is quite too complex for a simple change)
* Hover over the icon by the words `Active` 
* Observe the text `Transfer failed. Learn possible reasons why`:

<img width="255" alt="Screenshot 2024-09-02 at 12 11 43 PM" src="https://github.com/user-attachments/assets/471079aa-7fe8-4dc5-bb09-27960ef3f6d3">

* Click on the link
* Confirm that you are brought over to https://wordpress.com/support/domains/incoming-domain-transfer/#step-4-check-the-transfer-status

## Pre-merge Checklist
